src/components/hero.js

```diff
@@ -1,114 +1,251 @@
-import React from "react"
+import React, { useEffect, useRef, useState } from "react"
 import { Box, Text, Button, Flex } from "rebass"
-import { heroImg, heroBrandmark } from "../images"
+import { fabric } from "fabric"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import { faFacebookSquare } from "@fortawesome/free-brands-svg-icons"
+
+import theme from "../theme"
+import { heroImg, brandmarkPath } from "../images"
 import Container from "./container"
 
-const Hero = () => (
-  <Box
-    sx={{
-      position: "relative",
-      backgroundImage: `url("${heroImg}")`,
-      backgroundRepeat: "no-repeat",
-      backgroundPosition: "center",
-      backgroundSize: "cover",
-      width: "100%",
-      height: "100vh",
-      mt: "-105px",
-      mb: 4,
-    }}
-  >
+const getScaledHeroImage = (canvas, imageEl) => {
+  const heroImage = new fabric.Image(imageEl)
+  const canvasRatio = canvas.width / canvas.height
+  const imageRatio = imageEl.width / imageEl.height
+
+  if (imageRatio > canvasRatio) {
+    heroImage.scaleToHeight(canvas.height)
+  } else {
+    heroImage.scaleToWidth(canvas.width)
+  }
+
+  return heroImage
+}
+
+const rotatePath = (path, midpoint, { point, angle }) => {
```

Comment:
alot of these helper functions are outside the scope of the useEffect. Either move it there or register these as callbacks.

Reply:
Why though?

Reply:
https://github.com/facebook/react/issues/14920 it's exhaustive depths. Avoid closure pitfalls. why separate it if it won't be reused by other functions anyway?

Reply:
This is only an issue for functions that reference variables within the component scope. Everything is being passed in as parameters in this case, so it shouldn't be an issue! This is actually [one recommended approach to this problem](https://overreacted.io/a-complete-guide-to-useeffect/).

> First of all, if a function doesn’t use anything from the component scope, you can hoist it outside the component and then freely use it inside your effects:
>
> ![image](https://user-images.githubusercontent.com/7394331/75251776-4518cf00-5816-11ea-871b-8af4dc1f6034.png)

Regarding the separation, the functions were getting long and I didn't want to burden the component with the length of the code since they shouldn't be concerned with the implementation of the functions anyway.

Reply:
alright. i didn't notice it sooner that it's not inside the component. If I were you, I would create a helper lib to make it easy to do unit testing but shouldn't be a big deal.